### DISCUSSION TITLE
Reset axmol to default or checkin which axmol version I have to download it again.

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1733] [General]
Hello,
I messed up completely axmol because I did a word replacement that I thought would just affect my files, but it modified axmol's core too. So I would like to know if there is a way to revert axmol to the default. If not I am would like to download and reinstall the current axmol version I am working with but I don't where the version is displayed.
Thanks

**Comment by rh101** [comment]
If you want the latest version, and you don't use git for cloning the repository, then just do this:
![image](https://github.com/axmolengine/axmol/assets/8603230/43b620e4-d2b0-4ab6-b3ca-a23627fb0bfb)

If you want a tagged version, then click the tags option and download from there:
![image](https://github.com/axmolengine/axmol/assets/8603230/8872c3fa-bc01-4a8d-a4dd-1e5794cc2548)

**Reply by danialias** [reply]
I had the same replacement issue than @rudiHammad at least three times with Android Studio. The IDE suggested solutions for some problems change the source code of Axmol without warning... very inconvenient, because usually breaks everything. I wonder if some kind of writing protection could be activated to avoid it.

**Reply by rh101** [reply]
The functionality you need already exists.

In Android Studio, you would need to limit it to a directory or scope:
![image](https://github.com/axmolengine/axmol/assets/8603230/b0bc4871-03a8-49da-b5f7-cdd04b1611cb)

In the case of Visual studio, it's simple, since you can set it to apply changes to the current project, a directory, current document etc.:
![image](https://github.com/axmolengine/axmol/assets/8603230/adbf7821-332a-4487-9573-e479f537873e)

**Reply by danialias** [reply]
Didn't know about that option, thank you very much!

**Comment by rudiHammad** [comment]
Thanks @rh101 . I know about the tags but I was asking about how do I know which axmol version I currently have installed to know which one to download again. I have been looking at the files in axmol directory searching for "version" as key word, but I don't see any hint that tell me which one I have.

**Reply by rh101** [reply]
If you searched for "version", then the search result should have displayed "AX_VERSION", and if you follow that, you would end up at the right place.

The file you would end up at would be `core/axmolver.h`:
![image](https://github.com/axmolengine/axmol/assets/8603230/2adf01be-957f-4b84-96a1-204f53e1fe4d)

**Reply by rudiHammad** [reply]
ahhh, damn it...I was looking into the main directory opening the files such as setup.ps1, build.ps1, etc.. and searching for version in those files, not in windows. That was stupid... thanks.

**Reply by rh101** [reply]
Regardless of the above, you should really be using the latest stable release of whichever major version of the engine that you're using, and not just stick to the version you have, unless there is a very specific reason to do so.

**Reply by rh101** [reply]
> ahhh, damn it...I was looking into the main directory opening the files such as setup.ps1, build.ps1, etc.. and searching for version in those files, not in windows. That was stupid... thanks.

If ever in doubt, do a search in all files; basically a file filter of `"*.*"`

**Reply by rudiHammad** [reply]
Sure. The reason is that I just want to make sure that nothing breaks. Once I got it working, I'll update to the last.
What I find strange is that my version according to axmolver.h is #define AX_VERSION_STR "2.1.0" which is from jan 2024.
I was pretty sure that the last version I install way waaaay older than that...I guess I am remembering it wrong

**Reply by rudiHammad** [reply]
actually, I was checking some of our discord conversations when I was asking about something related to Tiled. That conversation is from 11/04/2023. Since then I haven't downloaded axmol, so I don't understand how it says version 2.1.0 from jan 2024.
Anyway, never mind.

**Reply by halx99** [reply]
Just use axmol commnad on your terminal: `axmol`

![image](https://github.com/axmolengine/axmol/assets/6977319/41e83585-65bc-4e82-adbf-6bbb287ec18e)

**Reply by rudiHammad** [reply]
Update to last axmol version. All working. Thanks guys
